### PR TITLE
Add adaptor class to lists to split them

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Added an adaptor class to the list patterns from Vanilla, which splits the list items over two columns.

## QA
- Run `gulp develop`
- Open example/_patterns_list-split.html
- Check that the lists are in two columns
- Modify the demo and try and break it
- Check on small screens that the list stacks in the correct order

## Details
This is a new class `is-split` which only works with list patterns.

Fixes https://github.com/ubuntudesign/vanilla-brochure-theme/issues/9

### CODE REVIEW

File: scss/_pattern_lists.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@mixin bt-p-lists {
+  @include bt-p-list--split;
+}
+
+// Adaptor class `is-split` to split the items of a list into two columns
+@mixin bt-p-list--split {
+  [class*='p-list'] {
+    &.is-split {
+
+      @include on-tablet-up {
```
Comment (barrymcgee) [inline scss/_pattern_lists.scss]: We should avoid using Shelves media query mixins and stick to using `@include (breakpoint-medium){}` to remain consistent with Vanilla and ensure patterns can be easily overridden. 

Breakpoint names/mixins should not also reference specific devices but should be abstract.

Comment (anthonydillon) [inline scss/_pattern_lists.scss]: done

**Comment by anthonydillon** [comment]
Updated Vanilla to the latest version, including your fix for the list-ticks.

- `rm -rf node_modules`
- `npm i`
- `gulp develop`
- Open the example in `example/_patterns_list-split.html`

**Comment by barrymcgee** [comment]
Great, thanks - looks good 👌🏻

**Pull request merged**